DependencyInjection/HttplugExtension.php

```diff
@@ -234,6 +237,40 @@ private function configureClient(ContainerBuilder $container, $name, array $argu
             $pluginClientOptions['debug_plugins'] = [new Reference($debugPluginServiceId)];
         }
 
+        if (array_key_exists('options', $arguments)) {
+            foreach ($arguments['options'] as $option => $value) {
+                switch ($option) {
+                    case 'default_host':
+                        $uriService = $serviceId.'.default_host_uri';
+                        $addHostPlugin = $serviceId.'.default_host_plugin';
+                        $this->createUri($container, $uriService, $value);
+                        $container
+                            ->register($addHostPlugin, AddHostPlugin::class)
+                            ->setPublic(false)
+                            ->addArgument(new Reference($uriService))
+                        ;
+                        if (!in_array($addHostPlugin, $plugins)) {
+                            $plugins[] = $addHostPlugin;
+                        }
+                        break;
+                    case 'force_host':
+                        $uriService = $serviceId.'.force_host_uri';
+                        $addHostPlugin = $serviceId.'.force_host_plugin';
+                        $this->createUri($container, $uriService, $value);
+                        $container
+                            ->register($addHostPlugin, AddHostPlugin::class)
+                            ->setPublic(false)
+                            ->addArgument(new Reference($uriService))
+                            ->addArgument(['replace' => true])
+                        ;
+                        if (!in_array($addHostPlugin, $plugins)) {
```

Comment:
by default, we append the AddHostPlugin to the very end of the chain, but you can explicitly set it in the plugins list. i will explain this in the documentation